<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Raquel Pinho" />


<title>RUV_Kallisto_tx_with_outliers.Rmd</title>

<script src="Goat_milkI_vs_No_milkI_Kallisto_tx_with_outliers_files/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="Goat_milkI_vs_No_milkI_Kallisto_tx_with_outliers_files/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="Goat_milkI_vs_No_milkI_Kallisto_tx_with_outliers_files/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="Goat_milkI_vs_No_milkI_Kallisto_tx_with_outliers_files/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="Goat_milkI_vs_No_milkI_Kallisto_tx_with_outliers_files/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="Goat_milkI_vs_No_milkI_Kallisto_tx_with_outliers_files/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="Goat_milkI_vs_No_milkI_Kallisto_tx_with_outliers_files/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="Goat_milkI_vs_No_milkI_Kallisto_tx_with_outliers_files/tocify-1.9.1/jquery.tocify.js"></script>
<script src="Goat_milkI_vs_No_milkI_Kallisto_tx_with_outliers_files/navigation-1.1/tabsets.js"></script>
<script src="Goat_milkI_vs_No_milkI_Kallisto_tx_with_outliers_files/navigation-1.1/codefolding.js"></script>
<link href="Goat_milkI_vs_No_milkI_Kallisto_tx_with_outliers_files/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="Goat_milkI_vs_No_milkI_Kallisto_tx_with_outliers_files/highlightjs-9.12.0/highlight.js"></script>
<link href="Goat_milkI_vs_No_milkI_Kallisto_tx_with_outliers_files/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="Goat_milkI_vs_No_milkI_Kallisto_tx_with_outliers_files/pagedtable-1.1/js/pagedtable.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>



<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = false;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}

.tocify-subheader {
  display: inline;
}
.tocify-subheader .tocify-item {
  font-size: 0.95em;
}

</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">RUV_Kallisto_tx_with_outliers.Rmd</h1>
<h4 class="author">Raquel Pinho</h4>
<h4 class="date">1 de maio de 2019</h4>

</div>


<div id="kallisto" class="section level1">
<h1><span class="header-section-number">1</span> Kallisto</h1>
</div>
<div id="data" class="section level1">
<h1><span class="header-section-number">2</span> Data</h1>
<p>This code was written using data from mhlz experiment 2016, where Malnourished and full fed animals were compared after the ingestion of regular milk, lysozyme-rich milk or no milk. The expression along the length of the intestinal tract were assessed by underlying distribution of RNA-seq data. In total we have seven groups and 4 or three animals per group.</p>
<p>I will first set the working directory to where output are going to be saved and choose the groups to be compared if it was not defined on the command line in the server, besides setting up the knitr options:</p>
<div id="setting-work-and-output-directory" class="section level2">
<h2><span class="header-section-number">2.1</span> Setting work and output directory</h2>
<pre class="r"><code>#args = commandArgs(trailingOnly=TRUE)
setwd(&quot;C:/Users/rpmello/Desktop/Data Analysis RNASeq/Projeto_Kallisto_DESeq_EdgeR&quot;)
#groups &lt;- c(args[1],args[2])
#groups&lt;-c(&quot;FFI&quot;,&quot;No_milkI&quot;)#colocar em ordem alfabÃ©tica e com a inicial da seÃ§Ã£o
dir.create(file.path(&quot;C:/Users/rpmello/Desktop/Data Analysis RNASeq/Projeto_Kallisto_DESeq_EdgeR/Output&quot;, sprintf(&quot;%s_vs_%s_Kallisto_with_outliers_R&quot;,groups[1], groups[2])), showWarnings = FALSE)
output_dir&lt;-file.path(&quot;C:/Users/rpmello/Desktop/Data Analysis RNASeq/Projeto_Kallisto_DESeq_EdgeR/Output&quot;, sprintf(&quot;%s_vs_%s_Kallisto_with_outliers_R&quot;,groups[1], groups[2]))
dir.create(file.path(output_dir, &quot;Figures&quot;), showWarnings = FALSE)
sprintf(&quot;Comparison of %s and %s.&quot;, groups[1], groups[2])</code></pre>
<pre><code>## [1] &quot;Comparison of Goat_milkI and No_milkI.&quot;</code></pre>
<pre class="r"><code>knitr::opts_chunk$set(echo = TRUE, results = &quot;asis&quot;, include = TRUE, fig.path = file.path(output_dir,&quot;Figures&quot;), fig.width = 11, fig.height = 8, fig.show = &quot;animated&quot;)</code></pre>
</div>
</div>
<div id="importing-results-from-kallisto" class="section level1">
<h1><span class="header-section-number">3</span> Importing results from Kallisto</h1>
<p>I will start creating a table of counts with all the samples.But for this I will have to import the counts data and the metadata.</p>
<div id="import-metadata-table" class="section level2">
<h2><span class="header-section-number">3.1</span> Import metadata table</h2>
<p>Import the metadata and the outliers based on Poisson distances</p>
<pre class="r"><code>setwd(&quot;C:/Users/rpmello/Desktop/Data Analysis RNASeq/Projeto_Kallisto_DESeq_EdgeR&quot;)
metadata&lt;- read.csv(&quot;./Data/metadata.csv&quot;)
tx2gene_k&lt;- read.csv(&quot;./Data/tx2gene_k.csv&quot;, header = T)</code></pre>
</div>
<div id="importing-abundance-data" class="section level2">
<h2><span class="header-section-number">3.2</span> Importing abundance data</h2>
<p>Create a variable for the path to the abundance files and insert it in the metadata table:</p>
<pre class="r"><code>path_to_files&lt;- sprintf(&quot;C:/Users/rpmello/Desktop/Data Analysis RNASeq/Kallisto/%s/%s_kaltranscboot_trim&quot;, &quot;results&quot;, metadata$SampleID)
path_to_files&lt;- cbind(path_to_files, metadata[,-c(1,3)])
selectMetadata_k &lt;- path_to_files[path_to_files$SampleID %in% metadata$SampleID[which(metadata$Group_sectio %in% groups)],]
selectMetadata_k&lt;-selectMetadata_k[order(selectMetadata_k$Group),]
Path_kallisto&lt;- file.path(selectMetadata_k$path_to_files,&quot;abundance.h5&quot;)
labels&lt;- c(paste0(selectMetadata_k$Group, 1:(length(selectMetadata_k$Group)/2)))
selectMetadata_k$Labels&lt;- labels
row.names(selectMetadata_k)&lt;-selectMetadata_k$Labels
head(selectMetadata_k)</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["path_to_files"],"name":[1],"type":["fctr"],"align":["left"]},{"label":["SampleID"],"name":[2],"type":["fctr"],"align":["left"]},{"label":["Pig"],"name":[3],"type":["fctr"],"align":["left"]},{"label":["Index"],"name":[4],"type":["int"],"align":["right"]},{"label":["Group"],"name":[5],"type":["fctr"],"align":["left"]},{"label":["Group_sectio"],"name":[6],"type":["fctr"],"align":["left"]},{"label":["Trial"],"name":[7],"type":["fctr"],"align":["left"]},{"label":["Time"],"name":[8],"type":["fctr"],"align":["left"]},{"label":["Section"],"name":[9],"type":["fctr"],"align":["left"]},{"label":["Diet"],"name":[10],"type":["fctr"],"align":["left"]},{"label":["Fed"],"name":[11],"type":["fctr"],"align":["left"]},{"label":["Ecoli"],"name":[12],"type":["fctr"],"align":["left"]},{"label":["SectionTimeFed"],"name":[13],"type":["fctr"],"align":["left"]},{"label":["TimeSectionFedChal"],"name":[14],"type":["fctr"],"align":["left"]},{"label":["Description"],"name":[15],"type":["fctr"],"align":["left"]},{"label":["TRT"],"name":[16],"type":["fctr"],"align":["left"]},{"label":["Collection"],"name":[17],"type":["fctr"],"align":["left"]},{"label":["RNA_ext"],"name":[18],"type":["int"],"align":["right"]},{"label":["Lib_const"],"name":[19],"type":["fctr"],"align":["left"]},{"label":["Suceptability"],"name":[20],"type":["fctr"],"align":["left"]},{"label":["Lane"],"name":[21],"type":["int"],"align":["right"]},{"label":["Weight_3Week"],"name":[22],"type":["dbl"],"align":["right"]},{"label":["..22"],"name":[23],"type":["lgl"],"align":["right"]},{"label":["RIN"],"name":[24],"type":["dbl"],"align":["right"]},{"label":["Labels"],"name":[25],"type":["chr"],"align":["left"]}],"data":[{"1":"C:/Users/rpmello/Desktop/Data Analysis RNASeq/Kallisto/results/index3L8_kaltranscboot_trim","2":"index3L8","3":"29_13","4":"3","5":"Goat_milk","6":"Goat_milkI","7":"mhLZ","8":"Wk5","9":"Ileum","10":"Mal","11":"Goat","12":"No","13":"Ileum_5MalGoat","14":"Wk5_Il_MalGoat","15":"29_13Il","16":"MalGoat","17":"6_17P","18":"13","19":"8_29","20":"NA","21":"8","22":"6.5","23":"NA","24":"9.1","25":"Goat_milk1"},{"1":"C:/Users/rpmello/Desktop/Data Analysis RNASeq/Kallisto/results/index4L7_kaltranscboot_trim","2":"index4L7","3":"31_2","4":"4","5":"Goat_milk","6":"Goat_milkI","7":"mhLZ","8":"Wk5","9":"Ileum","10":"Mal","11":"Goat","12":"No","13":"Ileum_5MalGoat","14":"Wk5_Il_MalGoat","15":"31_2Il","16":"MalGoat","17":"6_16P","18":"14","19":"8_29","20":"NA","21":"7","22":"8.1","23":"NA","24":"8.9","25":"Goat_milk2"},{"1":"C:/Users/rpmello/Desktop/Data Analysis RNASeq/Kallisto/results/index6L9_kaltranscboot_trim","2":"index6L9","3":"31_3","4":"6","5":"Goat_milk","6":"Goat_milkI","7":"mhLZ","8":"Wk5","9":"Ileum","10":"Mal","11":"Goat","12":"No","13":"Ileum_5MalGoat","14":"Wk5_Il_MalGoat","15":"31_3Il","16":"MalGoat","17":"6_16A","18":"14","19":"8_29","20":"NA","21":"9","22":"7.0","23":"NA","24":"9.0","25":"Goat_milk3"},{"1":"C:/Users/rpmello/Desktop/Data Analysis RNASeq/Kallisto/results/index8L10_kaltranscboot_trim","2":"index8L10","3":"31_13","4":"8","5":"Goat_milk","6":"Goat_milkI","7":"mhLZ","8":"Wk5","9":"Ileum","10":"Mal","11":"Goat","12":"No","13":"Ileum_5MalGoat","14":"Wk5_Il_MalGoat","15":"31_13Il","16":"MalGoat","17":"6_16A","18":"15","19":"8_29","20":"NA","21":"10","22":"9.2","23":"NA","24":"9.0","25":"Goat_milk4"},{"1":"C:/Users/rpmello/Desktop/Data Analysis RNASeq/Kallisto/results/index1L7_kaltranscboot_trim","2":"index1L7","3":"9_5","4":"1","5":"No_milk","6":"No_milkI","7":"mhLF","8":"Wk5","9":"Ileum","10":"Mal","11":"No","12":"No","13":"Ileum_5Mal","14":"Wk5_Il_MalNo","15":"9_5Il","16":"5Mal","17":"4_4","18":"18","19":"8_29","20":"NA","21":"7","22":"7.7","23":"NA","24":"9.3","25":"No_milk1"},{"1":"C:/Users/rpmello/Desktop/Data Analysis RNASeq/Kallisto/results/index3L9_kaltranscboot_trim","2":"index3L9","3":"6_13","4":"3","5":"No_milk","6":"No_milkI","7":"mhLF","8":"Wk5","9":"Ileum","10":"Mal","11":"No","12":"No","13":"Ileum_5Mal","14":"Wk5_Il_MalNo","15":"6_13Il","16":"5Mal","17":"4_4","18":"17","19":"8_29","20":"NA","21":"9","22":"5.4","23":"NA","24":"9.6","25":"No_milk2"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<pre class="r"><code>names(Path_kallisto)&lt;-labels</code></pre>
<div id="import-the-data-using-tximport" class="section level3">
<h3><span class="header-section-number">3.2.1</span> Import the data using tximport</h3>
<p>The resulting counts table is :</p>
<pre class="r"><code>head(as.data.frame(txikallisto$counts))</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":[""],"name":["_rn_"],"type":[""],"align":["left"]},{"label":["Goat_milk1"],"name":[1],"type":["dbl"],"align":["right"]},{"label":["Goat_milk2"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["Goat_milk3"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["Goat_milk4"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["No_milk1"],"name":[5],"type":["dbl"],"align":["right"]},{"label":["No_milk2"],"name":[6],"type":["dbl"],"align":["right"]},{"label":["No_milk3"],"name":[7],"type":["dbl"],"align":["right"]},{"label":["No_milk4"],"name":[8],"type":["dbl"],"align":["right"]}],"data":[{"1":"2237.0690","2":"2410.069","3":"1705.029","4":"2410.055","5":"834.000","6":"2013.056","7":"2936.056","8":"2085.000","_rn_":"ENSSSCG00000000002"},{"1":"2064.8956","2":"1840.028","3":"3706.352","4":"2531.433","5":"2896.035","6":"2725.737","7":"2525.139","8":"2648.602","_rn_":"ENSSSCG00000000003"},{"1":"220.0000","2":"314.000","3":"269.000","4":"327.000","5":"201.000","6":"264.000","7":"261.000","8":"338.000","_rn_":"ENSSSCG00000000005"},{"1":"515.0000","2":"160.000","3":"504.000","4":"320.000","5":"600.000","6":"627.000","7":"352.000","8":"327.000","_rn_":"ENSSSCG00000000006"},{"1":"535.0000","2":"632.000","3":"346.000","4":"583.000","5":"308.000","6":"566.000","7":"569.000","8":"516.000","_rn_":"ENSSSCG00000000007"},{"1":"899.7974","2":"1027.177","3":"2493.709","4":"2076.021","5":"1213.174","6":"1851.007","7":"1166.867","8":"1351.651","_rn_":"ENSSSCG00000000010"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
</div>
<div id="deseq2" class="section level1">
<h1><span class="header-section-number">4</span> DESeq2</h1>
<p>I will follow the DESeq2 tutorial found <a href="https://github.com/mikelove/asthma">here</a> and <a href="Angus_workshop">here</a> on the counts data from kallisto.</p>
<div id="removing-dependent-variables" class="section level3">
<h3><span class="header-section-number">4.0.1</span> Removing dependent variables</h3>
<p>I will remove the metadata that does not present a variation in this comparison or have all unique values in this comparison</p>
<pre class="r"><code>group&lt;-selectMetadata_k$Group
batch_effects&lt;-sapply(1:ncol(selectMetadata_k), function(i){length(unique(selectMetadata_k[,i]))!= 1 &amp; length(unique(selectMetadata_k[,i])) != length(selectMetadata_k[,i])})
batch_effects&lt;-selectMetadata_k[,batch_effects]
dependent&lt;-sapply(1:ncol(batch_effects), function(i){length(unique(batch_effects[1:4,i])) == 1 &amp; length(unique(batch_effects[5:8,i])) == 1})
independent&lt;-which(!dependent)
batch_effects_k&lt;-batch_effects[,independent]
batch_effects_k$Group&lt;-group
batch_effects_k</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":[""],"name":["_rn_"],"type":[""],"align":["left"]},{"label":["Index"],"name":[1],"type":["int"],"align":["right"]},{"label":["Trial"],"name":[2],"type":["fctr"],"align":["left"]},{"label":["Collection"],"name":[3],"type":["fctr"],"align":["left"]},{"label":["RNA_ext"],"name":[4],"type":["fctr"],"align":["left"]},{"label":["Lane"],"name":[5],"type":["fctr"],"align":["left"]},{"label":["RIN"],"name":[6],"type":["dbl"],"align":["right"]},{"label":["Litter"],"name":[7],"type":["fctr"],"align":["left"]},{"label":["Group"],"name":[8],"type":["fctr"],"align":["left"]}],"data":[{"1":"3","2":"mhLZ","3":"6_17P","4":"13","5":"8","6":"9.1","7":"29","8":"Goat_milk","_rn_":"Goat_milk1"},{"1":"4","2":"mhLZ","3":"6_16P","4":"14","5":"7","6":"8.9","7":"31","8":"Goat_milk","_rn_":"Goat_milk2"},{"1":"6","2":"mhLZ","3":"6_16A","4":"14","5":"9","6":"9.0","7":"31","8":"Goat_milk","_rn_":"Goat_milk3"},{"1":"8","2":"mhLZ","3":"6_16A","4":"15","5":"10","6":"9.0","7":"31","8":"Goat_milk","_rn_":"Goat_milk4"},{"1":"1","2":"mhLF","3":"4_4","4":"18","5":"7","6":"9.3","7":"9","8":"No_milk","_rn_":"No_milk1"},{"1":"3","2":"mhLF","3":"4_4","4":"17","5":"9","6":"9.6","7":"6","8":"No_milk","_rn_":"No_milk2"},{"1":"4","2":"mhLF","3":"4_3","4":"17","5":"8","6":"9.3","7":"7","8":"No_milk","_rn_":"No_milk3"},{"1":"5","2":"mhLZ","3":"6_17A","4":"14","5":"10","6":"8.4","7":"30","8":"No_milk","_rn_":"No_milk4"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<p>From now on I will use the manual from <a href="https://bioconductor.org/packages/release/bioc/vignettes/RUVSeq/inst/doc/RUVSeq.pdf">RUVSeq</a> and the manual from <a href="https://www.bioconductor.org/packages/devel/workflows/vignettes/rnaseqGene/inst/doc/rnaseqGene.html#using-ruv-with-deseq2">DESeq2</a></p>
</div>
<div id="deseq-object" class="section level2">
<h2><span class="header-section-number">4.1</span> DESeq object</h2>
<p>Creating a DESeqDataSet for the in this case we will use the design ~Group, because, for know we are only interested in the normalized counts to filter low counts and calculate the variation varible with RUV.</p>
<pre class="r"><code>library(DESeq2)
dds_k &lt;-DESeqDataSetFromTximport(txikallisto, batch_effects_k, as.formula(&quot;~Group&quot;))</code></pre>
<pre><code>## using counts and average transcript lengths from tximport</code></pre>
<pre class="r"><code>#It doesn&#39;t matter the formula now, because I am just interested in the counts, but I 
dds_k&lt;-estimateSizeFactors(dds_k)</code></pre>
<pre><code>## using &#39;avgTxLength&#39; from assays(dds), correcting for library size</code></pre>
<pre class="r"><code>dds_k&lt;-DESeq(dds_k)</code></pre>
<pre><code>## using pre-existing normalization factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates</code></pre>
<pre><code>## fitting model and testing</code></pre>
<div id="filtering-low-expressed-genes" class="section level3">
<h3><span class="header-section-number">4.1.1</span> Filtering low expressed genes</h3>
<p>I wil remove the genes if low counts, and I am being very stringent here, removing the ones tha have lower than 20 counts in all the samples and the ones that have less then 5 counts in at least two samples.</p>
<p>I will use the <a href="http://bioconductor.org/packages/release/bioc/manuals/EDASeq/man/EDASeq.pdf">EDASeq</a> package just to create an object were I can store the data and the metadata for multiple analysis.</p>
<pre class="r"><code>library(EDASeq)
library(RColorBrewer)
#creating a set with the expression dat
set_k&lt;-newSeqExpressionSet(DESeq2::counts(dds_k), phenoData= batch_effects_k)
set_k</code></pre>
<p>SeqExpressionSet (storageMode: lockedEnvironment) assayData: 22630 features, 8 samples element names: counts, normalizedCounts, offset protocolData: none phenoData sampleNames: Goat_milk1 Goat_milk2 … No_milk4 (8 total) varLabels: Index Trial … Group (8 total) varMetadata: labelDescription featureData: none experimentData: use ‘experimentData(object)’ Annotation:</p>
<pre class="r"><code>colors&lt;-brewer.pal(4,&quot;Set2&quot;)
idx  &lt;- rowSums(EDASeq::counts(set_k) &gt; 5) &gt;= 2
set_k  &lt;- set_k[idx, ]
filter&lt;-apply(EDASeq::counts(set_k),1,function(x){length(x[x&gt;5])&gt;2})
set_k &lt;-set_k[filter,]
sprintf(&quot;Removed %s genes with low counts from %s total genes&quot;, nrow(EDASeq::counts(dds_k))-nrow(EDASeq::counts(set_k)),nrow(DESeq2::counts(dds_k)))</code></pre>
<p>[1] “Removed 5894 genes with low counts from 22630 total genes”</p>
</div>
</div>
<div id="ruvseq" class="section level2">
<h2><span class="header-section-number">4.2</span> RUVSeq</h2>
<p>The number of sig DE genes with mod1 (~Group) was:</p>
<pre class="r"><code>res&lt;-results(dds_k)
sigres&lt;-res[!is.na(res$padj),]
sigres&lt;-sigres[sigres$padj&lt;0.1 &amp; sigres$log2FoldChange&gt;1| sigres$log2FoldChange&lt; -1,]
dim(sigres)</code></pre>
<p>[1] 590 6</p>
<div id="box-plot-of-the-counts-per-sample" class="section level3">
<h3><span class="header-section-number">4.2.1</span> Box plot of the counts per sample</h3>
<p>We can analyze the variation of the counts before normalization and reduced variation.</p>
<pre class="r"><code>plotRLE(set_k, outline =F, ylim=c(-2,2), col=colors[set_k$Group])</code></pre>
<div class="figure">
<img src="Figuresbox_plot_counts-1.png" />

</div>
<pre class="r"><code>plotPCA(set_k, col=colors[set_k$Group])</code></pre>
<div class="figure">
<img src="Figuresbox_plot_counts-2.png" />

</div>
</div>
<div id="upper-quartile-normalization" class="section level3">
<h3><span class="header-section-number">4.2.2</span> Upper quartile normalization</h3>
<p>The upper quartile normalization <a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-11-94">(UQ)</a> normalizes the data accounting for the most highly expressed transcripts, since there is a bias on the number of reads associated to them, i.e, more expresssed trancripts generate more reads and futhermore the number of reads on these highly expressed genes is proportional to the total number of reads in the sample (if samples are in individual lanes or in the same lane), which is not the case for me. <a href="http://vinaykmittal.blogspot.com/2013/10/fpkmrpkm-normalization-caveat-and-upper.html">Here</a>a more human explanation. And <a href="https://academic.oup.com/bib/article/14/6/671/189645">here</a> a more recent paper on RNA data normalization.</p>
<pre class="r"><code>set_n&lt;-betweenLaneNormalization(set_k,which = &quot;upper&quot;)
set_n$Group&lt;-factor(set_n$Group)
plotRLE(set_n, outline=F, col=colors[set_n$Group])</code></pre>
<div class="figure">
<img src="Figuresupper-1.png" />

</div>
<pre class="r"><code>plotPCA(set_n,col=colors[set_n$Group])</code></pre>
<div class="figure">
<img src="Figuresupper-2.png" />

</div>
</div>
<div id="using-empirical-control-genes" class="section level3">
<h3><span class="header-section-number">4.2.3</span> Using empirical control genes</h3>
<p>Since no spike in were used we will use the genes that have least changed between samples ( least significantly DE genes ) and use them as controls for the unwanted factors. So first we do a DE analysis and find the genes and then we do the normalization.</p>
</div>
<div id="empirical-controls-using-the-prior-de-analisis-using-deseq2" class="section level3">
<h3><span class="header-section-number">4.2.4</span> Empirical controls using the prior DE analisis using DESeq2</h3>
<p>Now we are going to use the DESeq2 to do DE analysis so we can pick the least DE genes as controls.In this case I will estimate only one unwanted variacen variable.</p>
<pre class="r"><code>#Using RUVSeq
library(RUVSeq)
not.sig &lt;- rownames(res)[which(res$pvalue &gt; .2)]
empirical &lt;- rownames(set_n)[ rownames(set_n) %in% not.sig ]
if(&quot;RDAVIDWebService&quot; %in% (.packages())){
  detach(&quot;package:RDAVIDWebService&quot;, unload=TRUE) 
}
set_k_n &lt;- RUVg(set_n, empirical, k=1)
pData(set_k_n)</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":[""],"name":["_rn_"],"type":[""],"align":["left"]},{"label":["Index"],"name":[1],"type":["int"],"align":["right"]},{"label":["Trial"],"name":[2],"type":["fctr"],"align":["left"]},{"label":["Collection"],"name":[3],"type":["fctr"],"align":["left"]},{"label":["RNA_ext"],"name":[4],"type":["fctr"],"align":["left"]},{"label":["Lane"],"name":[5],"type":["fctr"],"align":["left"]},{"label":["RIN"],"name":[6],"type":["dbl"],"align":["right"]},{"label":["Litter"],"name":[7],"type":["fctr"],"align":["left"]},{"label":["Group"],"name":[8],"type":["fctr"],"align":["left"]},{"label":["W_1"],"name":[9],"type":["dbl"],"align":["right"]}],"data":[{"1":"3","2":"mhLZ","3":"6_17P","4":"13","5":"8","6":"9.1","7":"29","8":"Goat_milk","9":"-0.27398666","_rn_":"Goat_milk1"},{"1":"4","2":"mhLZ","3":"6_16P","4":"14","5":"7","6":"8.9","7":"31","8":"Goat_milk","9":"-0.36212677","_rn_":"Goat_milk2"},{"1":"6","2":"mhLZ","3":"6_16A","4":"14","5":"9","6":"9.0","7":"31","8":"Goat_milk","9":"0.52536564","_rn_":"Goat_milk3"},{"1":"8","2":"mhLZ","3":"6_16A","4":"15","5":"10","6":"9.0","7":"31","8":"Goat_milk","9":"0.05659745","_rn_":"Goat_milk4"},{"1":"1","2":"mhLF","3":"4_4","4":"18","5":"7","6":"9.3","7":"9","8":"No_milk","9":"0.41927000","_rn_":"No_milk1"},{"1":"3","2":"mhLF","3":"4_4","4":"17","5":"9","6":"9.6","7":"6","8":"No_milk","9":"-0.18182358","_rn_":"No_milk2"},{"1":"4","2":"mhLF","3":"4_3","4":"17","5":"8","6":"9.3","7":"7","8":"No_milk","9":"-0.47173837","_rn_":"No_milk3"},{"1":"5","2":"mhLZ","3":"6_17A","4":"14","5":"10","6":"8.4","7":"30","8":"No_milk","9":"0.28844229","_rn_":"No_milk4"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<p>Using the unwanted variance variable in DESeq ( ~ W1 + Group).</p>
<pre class="r"><code>ddsruv &lt;- dds_k
ddsruv$W1 &lt;- set_k_n$W_1
design(ddsruv) &lt;- ~ W1 + Group
dds_k&lt;-ddsruv</code></pre>
<p>The resulting counts are displayed here:</p>
<pre class="r"><code>head(as.data.frame(DESeq2::counts(dds_k, normalized=TRUE)))</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":[""],"name":["_rn_"],"type":[""],"align":["left"]},{"label":["Goat_milk1"],"name":[1],"type":["dbl"],"align":["right"]},{"label":["Goat_milk2"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["Goat_milk3"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["Goat_milk4"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["No_milk1"],"name":[5],"type":["dbl"],"align":["right"]},{"label":["No_milk2"],"name":[6],"type":["dbl"],"align":["right"]},{"label":["No_milk3"],"name":[7],"type":["dbl"],"align":["right"]},{"label":["No_milk4"],"name":[8],"type":["dbl"],"align":["right"]}],"data":[{"1":"2397.2046","2":"2559.5300","3":"1470.6913","4":"2218.3861","5":"1108.5537","6":"1669.3720","7":"2661.8540","8":"2308.7576","_rn_":"ENSSSCG00000000002"},{"1":"2215.1849","2":"1929.9997","3":"3227.1142","4":"2326.9280","5":"3890.3286","6":"2255.4268","7":"2282.0203","8":"2927.5490","_rn_":"ENSSSCG00000000003"},{"1":"236.0185","2":"330.8935","3":"234.8168","4":"299.5219","5":"268.7696","6":"217.0635","7":"236.5057","8":"375.3263","_rn_":"ENSSSCG00000000005"},{"1":"552.9238","2":"168.3535","3":"437.5668","4":"293.8182","5":"803.4519","6":"526.3275","7":"316.9959","8":"358.7045","_rn_":"ENSSSCG00000000006"},{"1":"580.3825","2":"659.4220","3":"298.8848","4":"530.4098","5":"413.8568","6":"472.0827","7":"520.7936","8":"565.4805","_rn_":"ENSSSCG00000000007"},{"1":"957.3088","2":"1083.5023","3":"2179.2803","4":"1911.6540","5":"1626.8498","6":"1527.0605","7":"1048.6873","8":"1505.9049","_rn_":"ENSSSCG00000000010"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<div id="calculating-the-poisson-distance" class="section level2">
<h2><span class="header-section-number">4.3</span> Calculating the poisson distance</h2>
<div id="heatmap-of-the-log-transformed-counts" class="section level3">
<h3><span class="header-section-number">4.3.1</span> Heatmap of the log transformed counts</h3>
<pre class="r"><code>hc&lt;- hclust(poisd_k$dd)
heatmap.2(samplePoisDistMatrix_k, Rowv=as.dendrogram(hc), symm=T, trace=&quot;none&quot;, col= colors, labCol=F)</code></pre>
<p><img src="Figuresheatmap_pois_log-1.png" /><!-- --></p>
</div>
<div id="we-can-also-see-the-pca-plot" class="section level3">
<h3><span class="header-section-number">4.3.2</span> We can also see the PCA plot</h3>
<pre class="r"><code>library(ggplot2)
z &lt;-plotPCA(rlogdds_k, intgroup = c( &quot;Group&quot;))
z+ geom_text(aes(label = selectMetadata_k$Pig))</code></pre>
<p><img src="Figureslogcounts_PCA-1.png" /><!-- --></p>
</div>
</div>
<div id="de-analysis" class="section level2">
<h2><span class="header-section-number">4.4</span> DE analysis</h2>
<p>Using the desing, we have the following results:</p>
<p>Summary of the results:</p>
<pre class="r"><code>summary(res_k)</code></pre>
<p>out of 16206 with nonzero total read count adjusted p-value &lt; 0.1 LFC &gt; 0 (up) : 4, 0.025% LFC &lt; 0 (down) : 11, 0.068% outliers [1] : 0, 0% low counts [2] : 0, 0% (mean count &lt; 4) [1] see ‘cooksCutoff’ argument of ?results [2] see ‘independentFiltering’ argument of ?results</p>
<pre class="r"><code>res.sort&lt;- res_k[order(res_k$pvalue), ]</code></pre>
<div id="significant-de-genes" class="section level3">
<h3><span class="header-section-number">4.4.1</span> Significant DE genes</h3>
<p>Considering a padj &lt; 0.1 being significant we can subset the significant differentiated genes and we have this number of genes with padj &lt; 0.1 L2FC 1</p>
<pre class="r"><code>res_k&lt;- as.data.frame(res_k)
resSig_Kal &lt;- subset(res_k, padj&lt; 0.1)
resSig_Kal &lt;- subset(resSig_Kal, log2FoldChange &lt; -1 | log2FoldChange &gt; 1)
saveRDS(resSig_Kal, file = file.path(output_dir,sprintf(&quot;%s_vs_%s_resSig_Kal_with_outliers.RDS&quot;,groups[1], groups[2])))
dim(resSig_Kal)</code></pre>
<p>[1] 8 6</p>
<div id="we-can-be-more-stringent-and-put-the-log2foldchange-values--1.5-or-1.5" class="section level4">
<h4><span class="header-section-number">4.4.1.1</span> We can be more stringent and put the log2foldchange values &lt; -1.5 or &gt; 1.5</h4>
<pre class="r"><code>resSig_Kal1.5 &lt;- subset(resSig_Kal, log2FoldChange &lt; -1.5 | log2FoldChange &gt; 1.5)
dim(resSig_Kal1.5)</code></pre>
<p>[1] 5 6</p>
</div>
</div>
<div id="sorting-down-and-upregulated-genes" class="section level3">
<h3><span class="header-section-number">4.4.2</span> Sorting Down and upregulated genes</h3>
<p>We can use the log2foldchange display the most variable genes.</p>
<div id="the-top-downregulated-genes" class="section level4">
<h4><span class="header-section-number">4.4.2.1</span> The top downregulated genes:</h4>
<p>top downregulated genes ( based on l2fc) and total number of downregulated genes:</p>
<pre class="r"><code># sorting by the Log2FoldChange
head(as.data.frame(resSig_Kal[order(resSig_Kal$log2FoldChange), ]),30) # stronger downregulation</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":[""],"name":["_rn_"],"type":[""],"align":["left"]},{"label":["baseMean"],"name":[1],"type":["dbl"],"align":["right"]},{"label":["log2FoldChange"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["lfcSE"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["stat"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["pvalue"],"name":[5],"type":["dbl"],"align":["right"]},{"label":["padj"],"name":[6],"type":["dbl"],"align":["right"]}],"data":[{"1":"247.85310","2":"-4.826476","3":"1.1912809","4":"-4.051501","5":"5.089004e-05","6":"0.0687269932","_rn_":"ENSSSCG00000012871"},{"1":"101.37041","2":"-2.082408","3":"0.3585851","4":"-5.807291","5":"6.349156e-09","6":"0.0001028944","_rn_":"ENSSSCG00000020694"},{"1":"95.21088","2":"-1.530660","3":"0.3601937","4":"-4.249547","5":"2.142032e-05","6":"0.0347137690","_rn_":"ENSSSCG00000037751"},{"1":"333.58831","2":"-1.408579","3":"0.3563973","4":"-3.952272","5":"7.741283e-05","6":"0.0896108753","_rn_":"ENSSSCG00000015069"},{"1":"993.33241","2":"-1.274634","3":"0.2938651","4":"-4.337478","5":"1.441270e-05","6":"0.0259524691","_rn_":"ENSSSCG00000002383"},{"1":"568.26702","2":"1.279421","3":"0.2891828","4":"4.424264","5":"9.677178e-06","6":"0.0259524691","_rn_":"ENSSSCG00000027865"},{"1":"101.72472","2":"1.516753","3":"0.2863277","4":"5.297261","5":"1.175526e-07","6":"0.0009525288","_rn_":"ENSSSCG00000026943"},{"1":"40.26249","2":"2.850462","3":"0.6416866","4":"4.442140","5":"8.906860e-06","6":"0.0259524691","_rn_":"ENSSSCG00000026333"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<pre class="r"><code>#Getting the up regulated or down regulated genes 
downresSig_Kal&lt;-resSig_Kal[resSig_Kal$log2FoldChange &lt; -1,]
downresSig_Kal&lt;-downresSig_Kal[order(downresSig_Kal$log2FoldChange),]
dim(downresSig_Kal)</code></pre>
<p>[1] 5 6</p>
</div>
<div id="the-top-upregulated" class="section level4">
<h4><span class="header-section-number">4.4.2.2</span> The top upregulated:</h4>
<p>Top upregulated genes and the total number of upregulated genes:</p>
<pre class="r"><code>head(as.data.frame(resSig_Kal[order(-resSig_Kal$log2FoldChange), ]),30) #stronger upregulation</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":[""],"name":["_rn_"],"type":[""],"align":["left"]},{"label":["baseMean"],"name":[1],"type":["dbl"],"align":["right"]},{"label":["log2FoldChange"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["lfcSE"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["stat"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["pvalue"],"name":[5],"type":["dbl"],"align":["right"]},{"label":["padj"],"name":[6],"type":["dbl"],"align":["right"]}],"data":[{"1":"40.26249","2":"2.850462","3":"0.6416866","4":"4.442140","5":"8.906860e-06","6":"0.0259524691","_rn_":"ENSSSCG00000026333"},{"1":"101.72472","2":"1.516753","3":"0.2863277","4":"5.297261","5":"1.175526e-07","6":"0.0009525288","_rn_":"ENSSSCG00000026943"},{"1":"568.26702","2":"1.279421","3":"0.2891828","4":"4.424264","5":"9.677178e-06","6":"0.0259524691","_rn_":"ENSSSCG00000027865"},{"1":"993.33241","2":"-1.274634","3":"0.2938651","4":"-4.337478","5":"1.441270e-05","6":"0.0259524691","_rn_":"ENSSSCG00000002383"},{"1":"333.58831","2":"-1.408579","3":"0.3563973","4":"-3.952272","5":"7.741283e-05","6":"0.0896108753","_rn_":"ENSSSCG00000015069"},{"1":"95.21088","2":"-1.530660","3":"0.3601937","4":"-4.249547","5":"2.142032e-05","6":"0.0347137690","_rn_":"ENSSSCG00000037751"},{"1":"101.37041","2":"-2.082408","3":"0.3585851","4":"-5.807291","5":"6.349156e-09","6":"0.0001028944","_rn_":"ENSSSCG00000020694"},{"1":"247.85310","2":"-4.826476","3":"1.1912809","4":"-4.051501","5":"5.089004e-05","6":"0.0687269932","_rn_":"ENSSSCG00000012871"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<pre class="r"><code>upresSig_Kal&lt;-resSig_Kal[resSig_Kal$log2FoldChange &gt; 1,]
upresSig_Kal&lt;-upresSig_Kal[order(-upresSig_Kal$log2FoldChange),]
dim(upresSig_Kal)</code></pre>
<p>[1] 3 6</p>
</div>
</div>
<div id="list-of-genes-with-padj-0.1-l2fc--1-or-1." class="section level3">
<h3><span class="header-section-number">4.4.3</span> List of genes with padj &lt; 0.1, L2FC &lt; -1 or &gt; 1.</h3>
<p>Now, we can use BiomaRt to convert the ensembl ID into gene names.</p>
<p>For the downregulated genes ordered by padjusted and the total number of genes succefully converted by BiomaRt:</p>
<pre class="r"><code>down_genes_names&lt;-transform(merge(as.data.frame(downresSig_Kal),down_genes_names,by.x=0, by.y =&quot;ensembl_gene_id&quot;,all=TRUE), row.names=Row.names, Row.names=NULL)
down_genes_names_kal&lt;- down_genes_names[order(down_genes_names$padj),]
#The number of genes  we were able to find the gene name on BiomaRt
head(down_genes_names_kal, 30)</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":[""],"name":["_rn_"],"type":[""],"align":["left"]},{"label":["baseMean"],"name":[1],"type":["dbl"],"align":["right"]},{"label":["log2FoldChange"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["lfcSE"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["stat"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["pvalue"],"name":[5],"type":["dbl"],"align":["right"]},{"label":["padj"],"name":[6],"type":["dbl"],"align":["right"]},{"label":["external_gene_name"],"name":[7],"type":["chr"],"align":["left"]}],"data":[{"1":"101.37041","2":"-2.082408","3":"0.3585851","4":"-5.807291","5":"6.349156e-09","6":"0.0001028944","7":"DSG1","_rn_":"ENSSSCG00000020694"},{"1":"993.33241","2":"-1.274634","3":"0.2938651","4":"-4.337478","5":"1.441270e-05","6":"0.0259524691","7":"FOS","_rn_":"ENSSSCG00000002383"},{"1":"95.21088","2":"-1.530660","3":"0.3601937","4":"-4.249547","5":"2.142032e-05","6":"0.0347137690","7":"OTOF","_rn_":"ENSSSCG00000037751"},{"1":"247.85310","2":"-4.826476","3":"1.1912809","4":"-4.051501","5":"5.089004e-05","6":"0.0687269932","7":"FGF19","_rn_":"ENSSSCG00000012871"},{"1":"333.58831","2":"-1.408579","3":"0.3563973","4":"-3.952272","5":"7.741283e-05","6":"0.0896108753","7":"APOC3","_rn_":"ENSSSCG00000015069"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<pre class="r"><code>sum(down_genes_names$external_gene_name != &quot;&quot;)</code></pre>
<p>[1] 5</p>
<p>For the upregulated genes ordered by padjusted and the total number of genes succefully converted by BiomaRt:</p>
<pre class="r"><code>up_genes_names&lt;-transform(merge(as.data.frame(upresSig_Kal),up_genes_names,by.x=0, by.y =&quot;ensembl_gene_id&quot;,all=TRUE), row.names=Row.names, Row.names=NULL)
up_genes_names_kal&lt;- up_genes_names[order(up_genes_names$padj),]
#The number of genes  we were able to find the gene name on BiomaRt
sum(up_genes_names$external_gene_name != &quot;&quot;)</code></pre>
<p>[1] 3</p>
<pre class="r"><code>head(up_genes_names_kal,30)</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":[""],"name":["_rn_"],"type":[""],"align":["left"]},{"label":["baseMean"],"name":[1],"type":["dbl"],"align":["right"]},{"label":["log2FoldChange"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["lfcSE"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["stat"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["pvalue"],"name":[5],"type":["dbl"],"align":["right"]},{"label":["padj"],"name":[6],"type":["dbl"],"align":["right"]},{"label":["external_gene_name"],"name":[7],"type":["chr"],"align":["left"]}],"data":[{"1":"101.72472","2":"1.516753","3":"0.2863277","4":"5.297261","5":"1.175526e-07","6":"0.0009525288","7":"MRAP2","_rn_":"ENSSSCG00000026943"},{"1":"40.26249","2":"2.850462","3":"0.6416866","4":"4.442140","5":"8.906860e-06","6":"0.0259524691","7":"GTSF1","_rn_":"ENSSSCG00000026333"},{"1":"568.26702","2":"1.279421","3":"0.2891828","4":"4.424264","5":"9.677178e-06","6":"0.0259524691","7":"GALC","_rn_":"ENSSSCG00000027865"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<p>I will use the list to annotate the genes using the Biomart in Ensembl.</p>
<pre class="r"><code>#I will get the IDS and will use the following code
resSig_Kal&lt;- resSig_Kal[order(resSig_Kal$padj), ]
gene_names_removed_outliers_kal&lt;- getBM(attributes = c(&quot;ensembl_gene_id&quot;,&quot;external_gene_name&quot;), filters= &quot;ensembl_gene_id&quot;, values = rownames(resSig_Kal), mart = Pig_ensembl)
gene_names_removed_outliers_kal&lt;-transform(merge(as.data.frame(resSig_Kal),gene_names_removed_outliers_kal,by.x=0, by.y =&quot;ensembl_gene_id&quot;,all=TRUE), row.names=Row.names, Row.names=NULL)
gene_names_removed_outliers_kal&lt;-as.data.frame(gene_names_removed_outliers_kal[order(gene_names_removed_outliers_kal$padj),])
head(gene_names_removed_outliers_kal)</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":[""],"name":["_rn_"],"type":[""],"align":["left"]},{"label":["baseMean"],"name":[1],"type":["dbl"],"align":["right"]},{"label":["log2FoldChange"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["lfcSE"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["stat"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["pvalue"],"name":[5],"type":["dbl"],"align":["right"]},{"label":["padj"],"name":[6],"type":["dbl"],"align":["right"]},{"label":["external_gene_name"],"name":[7],"type":["chr"],"align":["left"]}],"data":[{"1":"101.37041","2":"-2.082408","3":"0.3585851","4":"-5.807291","5":"6.349156e-09","6":"0.0001028944","7":"DSG1","_rn_":"ENSSSCG00000020694"},{"1":"101.72472","2":"1.516753","3":"0.2863277","4":"5.297261","5":"1.175526e-07","6":"0.0009525288","7":"MRAP2","_rn_":"ENSSSCG00000026943"},{"1":"993.33241","2":"-1.274634","3":"0.2938651","4":"-4.337478","5":"1.441270e-05","6":"0.0259524691","7":"FOS","_rn_":"ENSSSCG00000002383"},{"1":"40.26249","2":"2.850462","3":"0.6416866","4":"4.442140","5":"8.906860e-06","6":"0.0259524691","7":"GTSF1","_rn_":"ENSSSCG00000026333"},{"1":"568.26702","2":"1.279421","3":"0.2891828","4":"4.424264","5":"9.677178e-06","6":"0.0259524691","7":"GALC","_rn_":"ENSSSCG00000027865"},{"1":"95.21088","2":"-1.530660","3":"0.3601937","4":"-4.249547","5":"2.142032e-05","6":"0.0347137690","7":"OTOF","_rn_":"ENSSSCG00000037751"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<pre class="r"><code>dim(gene_names_removed_outliers_kal)</code></pre>
<p>[1] 8 7</p>
</div>
</div>
<div id="diagnostic-plots" class="section level2">
<h2><span class="header-section-number">4.5</span> Diagnostic plots</h2>
<p>To plot the counts of specific genes we can use the name of the gene or a a specific value related to it</p>
<div id="volcano-plot" class="section level3">
<h3><span class="header-section-number">4.5.1</span> Volcano plot</h3>
<pre class="r"><code>plot(res_k$log2FoldChange, -log(res_k$pvalue),main =&quot;Volcano plot Kallisto data from DESeq2&quot;, xlab= &quot;Log2FoldChange&quot;, ylab = &quot;-log(Pvalue)&quot;, pch = 21, col=ifelse(res_k$pvalue &lt;0.001,&quot;red&quot;,&quot;black&quot;), xlim= c(-5,5), ylim =c(0,25), abline(v=c(-2,2)))</code></pre>
<p><img src="Figuresvolcano_plot-1.png" /><!-- --></p>
</div>
<div id="plotting-the-lowest-padj-gene" class="section level3">
<h3><span class="header-section-number">4.5.2</span> Plotting the lowest padj gene</h3>
<pre class="r"><code>topGene&lt;- row.names(resSig_Kal[which.min(resSig_Kal$padj),])
plotCounts(dds_k,gene= topGene,main = gene_names_removed_outliers_kal$external_gene_name[1], intgroup=&quot;Group&quot;)</code></pre>
<p><img src="Figureslowest_padj-1.png" /><!-- --></p>
</div>
<div id="to-do-the-ma-plot" class="section level3">
<h3><span class="header-section-number">4.5.3</span> to do the MA plot</h3>
<pre class="r"><code>plotMA(res_k, ylim=c(-20,20))</code></pre>
<p><img src="FiguresMAplot-1.png" /><!-- --></p>
</div>
<div id="to-plot-dispersion." class="section level3">
<h3><span class="header-section-number">4.5.4</span> To plot dispersion.</h3>
<p>(within group variability of each gene )</p>
<pre class="r"><code>plotDispEsts(dds_k)</code></pre>
<p><img src="Figuresdispersion_plot-1.png" /><!-- --></p>
</div>
<div id="histogram-of-pvalues" class="section level3">
<h3><span class="header-section-number">4.5.5</span> histogram of pvalues</h3>
<pre class="r"><code>hist(res_k$pvalue[res_k$baseMean&gt;1], breaks=20, col=&quot;grey50&quot;, border=&quot;white&quot;)</code></pre>
<p><img src="Figureshist_pvalue-1.png" /><!-- --></p>
</div>
<div id="clustering" class="section level3">
<h3><span class="header-section-number">4.5.6</span> Clustering</h3>
<p>of rlog counts of the top 50 DE genes</p>
<pre class="r"><code>heatmap.2(mat,Colv=F, trace=&quot;none&quot;, col=colors, ColSideColors=sidecols, labRow=F,scale=&quot;row&quot;)</code></pre>
<pre><code>## Warning in heatmap.2(mat, Colv = F, trace = &quot;none&quot;, col = colors,
## ColSideColors = sidecols, : Discrepancy: Colv is FALSE, while dendrogram is
## `both&#39;. Omitting column dendogram.</code></pre>
<p><img src="Figuresheatmap_kal-1.png" /><!-- --></p>
</div>
<div id="clustering-of-normalized-counts" class="section level3">
<h3><span class="header-section-number">4.5.7</span> Clustering of normalized counts</h3>
<p>150 more variable DE genes (if DE genes &lt; 150, all genes)</p>
</div>
</div>
</div>
<div id="prepare-for-the-heatmap" class="section level1">
<h1><span class="header-section-number">5</span> Prepare for the heatmap</h1>
<pre class="r"><code># Get some nicer colours
mypalette &lt;- brewer.pal(11,&quot;RdYlBu&quot;)

morecols &lt;- colorRampPalette(mypalette)

# Load library for heatmap
heatmap.2(highly_variable_norm_counts, col=rev(morecols(50)), trace=&quot;none&quot;,
              main=&quot;Most variable genes across samples&quot;, scale=&quot;row&quot;,
              labCol = labels)</code></pre>
<p><img src="Figurestpm_heatmap-1.png" /><!-- --></p>
<pre class="r"><code>#To save the gene lists in both the cluster( 100 more variable genes in the resSig_Salmon2 table) and in the resSig_Salmon2 so I can used the transcripts to annotation and the other downstream application like panther or DAVID.</code></pre>
<p>Cluster with all the DE genes using TPM</p>
<pre class="r"><code>DE_counts&lt;- tpmDE
heatmap.2(DE_counts, col=rev(morecols(50)), trace=&quot;none&quot;,
              main=&quot;Most variable genes across samples&quot;, scale=&quot;row&quot;,
              labCol = labels)</code></pre>
<p><img src="Figurescluster_tpmde_all1-1.png" /><!-- --></p>
<div id="pathway-analysis" class="section level2">
<h2><span class="header-section-number">5.1</span> Pathway Analysis</h2>
<p>I will use the lists to do the Pathway analysis with ReactomePA and DAVIDWebService</p>
<p>Fist, geting the GO terms for the DEgenes (padj,0.1, L2FC 1).</p>
<p>For that, I will first connect to the DAVID brownser:</p>
<p>To see the percentage of the genes that were identified by the DAVID</p>
<pre class="r"><code>DAVID_results_DESeq_Kal$inDavid</code></pre>
<p>[1] 0.875</p>
<p>And then you can create the results charts from DAVID and filter them by biological process:</p>
<pre class="r"><code>DAVID_results_chart_kal&lt;- getFunctionalAnnotationChart(david)
#If I want to create a file with the chart information I would use the following code:
#DAVID_results_chart&lt;-DAVIDFunctionalAnnotationChartFile(DAVID_results_chart)
DAVID_results_chart_DESeq_kal_BP&lt;-DAVID_results_chart_kal[DAVID_results_chart_kal$Category == &quot;GOTERM_BP_DIRECT&quot; &amp; DAVID_results_chart_kal$PValue &lt; 0.05,]</code></pre>
<p>We can plot the 5 most enriched pathways and the genes associated to those pathways</p>
<pre class="r"><code>if(dim(DAVID_results_chart_DESeq_kal_BP)[1]&gt;0){
plot2D(DAVIDFunctionalAnnotationChart(DAVID_results_chart_DESeq_kal_BP[1:5,]),
       color=c(&quot;FALSE&quot;=&quot;black&quot;, &quot;TRUE&quot;=&quot;green&quot;))
# We can see the top 5 GO pathways for biological process, and note how the evidence is true in our data for each of the genes in the pathway( if it is present on our sample or not)
}else{
  print(&quot;No significant term found&quot;)
}</code></pre>
<p>[1] “No significant term found”</p>
<div id="godag" class="section level3">
<h3><span class="header-section-number">5.1.1</span> GODag</h3>
<p>We can create a godag object from the functional chart</p>
<pre class="r"><code>DAVIDGODag_DESeq_kal&lt;-DAVIDGODag(DAVIDFunctionalAnnotationChart(DAVID_results_chart_DESeq_kal_BP), type =&quot;BP&quot;,pvalueCutoff=.001)
#I can isolate the significant ones by doing a dataframe but the resulting plot is the same 
#sig_DAVID_RES_CHART&lt;- DAVID_results_chart_BP[DAVID_results_chart_BP$PValue &lt;0.001,]
#sig_DAVID_RES_GODAG&lt;- DAVIDGODag(DAVIDFunctionalAnnotationChart(sig_DAVID_RES_CHART), type =&quot;BP&quot;,pvalueCutoff=.001)
#we used a cutoff of 0.001 f or the pvalue. And now we can inspect the Dag object</code></pre>
<p>We can get the most significant (p &lt;p=0.0001) categories (GO ID) and the summary of the analysis</p>
<pre class="r"><code>sigCategories(DAVIDGODag_DESeq_kal, p=0.0001)</code></pre>
<p>NULL</p>
<pre class="r"><code>summary(DAVIDGODag_DESeq_kal)</code></pre>
<pre><code>## Warning: No results met the specified criteria. Returning 0-row data.frame</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["GOBPID"],"name":[1],"type":["chr"],"align":["left"]},{"label":["Pvalue"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["Count"],"name":[3],"type":["int"],"align":["right"]},{"label":["Size"],"name":[4],"type":["int"],"align":["right"]},{"label":["Term"],"name":[5],"type":["chr"],"align":["left"]}],"data":[],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<pre class="r"><code>show(DAVIDGODag_DESeq_kal)</code></pre>
<p>Gene to GO BP test for over-representation 0 GO BP ids tested (0 have p &lt; 0.001) Selected gene set size: 0</p>
<pre><code>## Warning in max(popTotals(r)): no non-missing arguments to max; returning -
## Inf</code></pre>
<pre><code>Gene universe size: -Inf 
Annotation package: DAVID.website </code></pre>
<pre class="r"><code># we can see by the FDR column that only the first 3 pathway are significant</code></pre>
<p>And plot the terms in a graph form:</p>
<pre class="r"><code>plotGOTermGraph(g=goDag(DAVIDGODag_DESeq_kal), r= DAVIDGODag_DESeq_kal, max.nchar= 20, node.shape=&quot;ellipse&quot;,node.colors=c(sig=&quot;red&quot;, not=&quot;white&quot;))</code></pre>
<pre><code>## Warning in .local(x, y, ...): graph has zero nodes; cannot layout</code></pre>
</div>
<div id="using-goplot-to-visualize-the-goterm-from-david" class="section level3">
<h3><span class="header-section-number">5.1.2</span> Using GOplot to visualize the GOterm from DAVID</h3>
<p>First, we have to put the data in the format required by the package.</p>
<pre class="r"><code>#for the expression data I will use the counts from the DGElist from the importated salmon data.
cts_k&lt;-DESeq2::counts(dds_k, normalized=TRUE)
saveRDS(cts_k,file = file.path(output_dir,sprintf(&quot;%s_vs_%s_counts_TPM_Kal_with_outliers.RDS&quot;,groups[1], groups[2])))
eset &lt;- cts_k
genelist&lt;- as.data.frame(resSig_Kal)
names(genelist)[2]&lt;-&quot;logFC&quot;
genelist$ID&lt;-as.factor(rownames(genelist))
genelist&lt;-genelist[,c(7,2,6)]
david_fun_k&lt;- DAVID_results_chart_DESeq_kal_BP
Term&lt;- gsub(&quot;^.*?~&quot;,&quot;&quot;,DAVID_results_chart_DESeq_kal_BP$Term)
ID&lt;- gsub(&quot;~.*&quot;, &quot;&quot;, DAVID_results_chart_DESeq_kal_BP$Term)
#removing the GOID of the term. the(^) is the beginig of the string, (.) matching any characther, (*)repeated multiple times until get to ~, and plepace it by nothing &quot;&quot;.The ? makes it to match until the first ~ , if there were two ~. 
david_fun_k$Term&lt;-as.factor(Term)
david_fun_k$ID&lt;- as.factor(ID)
david_fun_k$Category &lt;-as.factor(c(rep(&quot;BP&quot;, length(david_fun_k$Category) )))
if(dim(david_fun_k)[1]&gt;0){
colnames(david_fun_k)[13]&lt;-&quot;adj_pval&quot;
david_fun_k &lt;- david_fun_k[,c(1,14,2,6,13)]
head(david_fun_k)
#Now I will choose the first 5 more significant processes
process&lt;-as.character(david_fun_k[order(david_fun_k$adj_pval),3])[1:5]
dataGOplot_kal&lt;-list(eset = eset,genelist = genelist,david = as.data.frame(david_fun_k),process = process)
}else{
  print(&quot;Functional chart empty&quot;)
}</code></pre>
<p>[1] “Functional chart empty”</p>
<p>. Now, that we have the data in the right format let’s plot it using GOplot.</p>
<div id="plot-the-pathway-colered-by-z-score-significance" class="section level4">
<h4><span class="header-section-number">5.1.2.1</span> Plot the pathway colered by z-score significance</h4>
<pre class="r"><code>if(dim(david_fun_k)[1]&gt;0 &amp; length(unique(circ_k$zscore)) &gt; 1){
print(GOBar(circ_k))}else {
  print(&quot;No significant terms&quot;)
}</code></pre>
<p>[1] “No significant terms”</p>
</div>
<div id="bubble-plot-of-pathways" class="section level4">
<h4><span class="header-section-number">5.1.2.2</span> Bubble plot of pathways</h4>
<pre class="r"><code>if(min(circ_k$adj_pval) &lt; 0.05){
print(GOBubble(circ_k, labels = 1))} else {
  print(&quot;No significant terms&quot;)
}</code></pre>
<p><img src="FiguresGOBubble-1.png" /><!-- -->TableGrob (1 x 2) “arrange”: 2 grobs z cells name grob 1 1 (1-1,1-1) arrange gtable[layout] 2 2 (1-1,2-2) arrange gtable[colhead-fg]</p>
</div>
<div id="plot-significant-enriched-pathways-and-de-genes" class="section level4">
<h4><span class="header-section-number">5.1.2.3</span> Plot significant enriched pathways and DE Genes</h4>
<p>This plot is interesting because it shows which genes are up and down on the pathway</p>
<pre class="r"><code>if(dim(david_fun_k)[1]&gt; 0 &amp; length(unique(circ_k$zscore)) &gt; 1){
nsub&lt;- ifelse(dim(circ_k[!duplicated(circ_k$term), ])[1] &gt; 10, 10, dim(circ_k[!duplicated(circ_k$term), ])[1]);
print(GOCircle(circ_k,nsub = nsub))} else {
  print(&quot;No significant terms&quot;)
}</code></pre>
<p>[1] “No significant terms”</p>
<p>I can also plot the relationship between the DE genes and the most enriched pathways</p>
<pre class="r"><code>if(dim(david_fun_k)[1]&gt;0){
subData &lt;- subset(circ_k, term %in% process);
genes&lt;-subset(genelist, ID %in% subData$genes);
All_genes&lt;- getBM(attributes = c(&quot;ensembl_gene_id&quot;,&quot;external_gene_name&quot;), filters= &quot;ensembl_gene_id&quot;, values = row.names(res_k), mart = Pig_ensembl);
names_subData&lt;- subset(All_genes, ensembl_gene_id %in% subData$genes);
names_subData$external_gene_name&lt;- ifelse(names_subData$external_gene_name == &quot;&quot;, names_subData$ensembl_gene_id, names_subData$external_gene_name);
chord_k &lt;- chord_dat(circ_k, genes = genes, process = dataGOplot_kal$process);
chord_names&lt;-as.data.frame(cbind(chord_k, names_subData[,&quot;external_gene_name&quot;][match(rownames(chord_k), names_subData$ensembl_gene_id)]));
row.names(chord_k)&lt;-chord_names$V7;
chord_k&lt;-chord_k[,colSums(chord_k != 0) &gt; 0];
print(GOChord(chord_k));
chord&lt;-chord_k;
terms&lt;-dataGOplot_kal$process[!is.na(dataGOplot_kal$process)]
if(length(terms)&gt;2){
print(GOCluster(circ_k, dataGOplot_kal$process, clust.by = &quot;term&quot;, lfc.col = c(&#39;darkgoldenrod1&#39;,&#39;black&#39;,&#39;cyan1&#39;)))}}else{
  print(&quot;No significant term found!&quot;)
}</code></pre>
<p>[1] “No significant term found!”</p>
<pre class="r"><code>#the GOCluster indicated in the outer ring the pathway that the gene is associated to, the inner ring shows the log2fc of the gene and the clustering the gene by category, euclidean average </code></pre>
</div>
</div>
<div id="reactome-pathways" class="section level3">
<h3><span class="header-section-number">5.1.3</span> Reactome pathways</h3>
<p>Using Biomart to convert the genes names into entrez ID (human) so I can use ReactomePA to do the pathway analysis</p>
<p>First table converting all the genes to entrex id and then filtering by padj &lt; 0.1</p>
<pre class="r"><code>allentrez&lt;- getBM(attributes = c(&quot;external_gene_name&quot;, &quot;entrezgene&quot;), filters= &quot;external_gene_name&quot;, values = All_genes$external_gene_name, mart = Human) </code></pre>
<pre><code>## Batch submitting query [=&gt;------------------------] 6% eta: 1m Batch
## submitting query [=&gt;------------------------] 9% eta: 1m Batch submitting
## query [==&gt;-----------------------] 12% eta: 1m Batch submitting
## query [===&gt;----------------------] 15% eta: 1m Batch submitting
## query [====&gt;---------------------] 18% eta: 1m Batch submitting
## query [=====&gt;--------------------] 21% eta: 1m Batch submitting
## query [=====&gt;--------------------] 24% eta: 1m Batch submitting
## query [======&gt;-------------------] 27% eta: 1m Batch submitting
## query [=======&gt;------------------] 30% eta: 1m Batch submitting
## query [========&gt;-----------------] 33% eta: 1m Batch submitting
## query [========&gt;-----------------] 36% eta: 1m Batch submitting
## query [=========&gt;----------------] 39% eta: 1m Batch submitting
## query [==========&gt;---------------] 42% eta: 1m Batch submitting
## query [===========&gt;--------------] 45% eta: 1m Batch submitting
## query [============&gt;-------------] 48% eta: 1m Batch submitting
## query [============&gt;-------------] 52% eta: 1m Batch submitting
## query [=============&gt;------------] 55% eta: 1m Batch submitting
## query [==============&gt;-----------] 58% eta: 1m Batch submitting
## query [===============&gt;----------] 61% eta: 48s Batch submitting
## query [================&gt;---------] 64% eta: 44s Batch submitting
## query [================&gt;---------] 67% eta: 40s Batch submitting
## query [=================&gt;--------] 70% eta: 38s Batch submitting
## query [==================&gt;-------] 73% eta: 34s Batch submitting
## query [===================&gt;------] 76% eta: 30s Batch submitting
## query [===================&gt;------] 79% eta: 26s Batch submitting
## query [====================&gt;-----] 82% eta: 22s Batch submitting
## query [=====================&gt;----] 85% eta: 18s Batch submitting
## query [======================&gt;---] 88% eta: 15s Batch submitting
## query [=======================&gt;--] 91% eta: 11s Batch submitting
## query [=======================&gt;--] 94% eta: 7s Batch submitting
## query [========================&gt;-] 97% eta: 4s Batch submitting query
## [==========================] 100% eta: 0s</code></pre>
<pre class="r"><code>l2fc&lt;- as.data.frame(res_k[,c(2,6)])
All_genes_entrez_rank&lt;-merge(l2fc, All_genes, by.x = &quot;row.names&quot;,by.y = &quot;ensembl_gene_id&quot;)
All_genes_entrez_rank&lt;- merge(All_genes_entrez_rank, allentrez, by= &quot;external_gene_name&quot;)
all_genes_list&lt;-All_genes_entrez_rank$log2FoldChange
names(all_genes_list)&lt;-All_genes_entrez_rank$entrezgene
all_genes_list&lt;-sort(all_genes_list, decreasing = T)
DE_genes_entrez_rank&lt;- All_genes_entrez_rank[All_genes_entrez_rank$padj &lt; 0.1 &amp; !is.na(All_genes_entrez_rank$padj),]
DE_genes_list_k&lt;- DE_genes_entrez_rank$log2FoldChange
names(DE_genes_list_k)&lt;- DE_genes_entrez_rank$entrezgene
DE_genes_list_k&lt;- sort(DE_genes_list_k, decreasing = T)</code></pre>
<div id="using-reactome-to-analize-subsetted-from-all-genes" class="section level4">
<h4><span class="header-section-number">5.1.3.1</span> Using reactome to analize subsetted from all genes</h4>
<p>Plots:</p>
<pre class="r"><code>if(!is.null(x)){
if(min(x@result$p.adjust)&lt;0.05){
sprintf(&quot;Barplot for the most enriched reactome pathway for Kallisto/DESeq analysis for comparison %s vs %s&quot;, groups[1],groups[2]);
print(barplot(x, showCategory = 30));
sprintf(&quot;Dotplot for the most enriched reactome pathway for Kallisto/DESeq analysis for comparison %s vs %s&quot;, groups[1],groups[2]);
print(dotplot(x, showCategory=15));
sprintf(&quot;Enrichment map plot for the most enriched reactome pathway for Kallisto/DESeq analysis for comparison %s vs %s&quot;, groups[1],groups[2]);
print(emapplot(x, showCategory = 15, color = &quot;p.adjust&quot;,layout = &quot;kk&quot;,colorEdge = TRUE)); 
print(cnetplot(x,showCategory = 5, categorySize = &quot;pvalue&quot;, foldChange = DE_genes_entrez_rank$log2FoldChange, node_label= F, colorEdge = T))} else {
  print(&quot;no enriched term found&quot;)
}}</code></pre>
<p>[1] “no enriched term found”</p>
<p>Ploting the GSEA ( gene set enrichment analysis): where the green line is the enrichment score, and the peak of the line is the emrichment score of the gene set. the gene following the peak are the one that more influenced the score. the lines are where the genes are in the ranked list and the bigger the line greater the correlation between the gene and the phenotype. The gene above zero are related to the first phenotype and the ones bellow zeros to the other phenotype.</p>
<pre class="r"><code>if(nrow(y) &gt; 0){
print(gseaplot(y, geneSetID =(as.data.frame(y))[1,1],title = as.character((as.data.frame(y))[1,2])));
print(cnetplot(y, categorySize = &quot;pvalue&quot;, foldChange = DE_genes_entrez_rank$log2FoldChange,colorEdge = TRUE,node_label= FALSE));
print(emapplot(y, showCategory = 15, color = &quot;p.adjust&quot;,layout = &quot;kk&quot;));
}
#DEunique&lt;- DE_genes_list[!duplicated(names(DE_genes_list))]</code></pre>
<div id="reactome-of-the-upregulated-genes" class="section level5">
<h5><span class="header-section-number">5.1.3.1.1</span> Reactome of the upregulated genes</h5>
<pre class="r"><code>if(!is.null(x)){
if(min(x@result$p.adjust)&lt;0.05){
sprintf(&quot;Barplot for the most enriched reactome pathway for Kallisto/DESeq upregulated genes analysis for comparison %s vs %s&quot;, groups[1],groups[2]);
print(barplot(x, showCategory = 30));
sprintf(&quot;Dotplot for the most enriched reactome pathway for Kallisto/DESeq  upregulated genes analysis for comparison %s vs %s&quot;, groups[1],groups[2]);
print(dotplot(x, showCategory=15));
sprintf(&quot;Enrichment map plot for the most enriched reactome pathway for Kallisto/DESeq upregulated genes analysis for comparison %s vs %s&quot;, groups[1],groups[2]);
print(emapplot(x, showCategory = 15, color = &quot;p.adjust&quot;,layout = &quot;kk&quot;,colorEdge = T) );
print(cnetplot(x,showCategory = 5, categorySize = &quot;pvalue&quot;, foldChange = up_genes_list_k, node_label= F, colorEdge = T))}
}</code></pre>
<pre><code>## wrong orderBy parameter; set to default `orderBy = &quot;x&quot;`</code></pre>
<p><img src="Figuresreactomeup_plots-1.png" /><img src="Figuresreactomeup_plots-2.png" /><img src="Figuresreactomeup_plots-3.png" /><img src="Figuresreactomeup_plots-4.png" /></p>
<p>GSEA for the upregulated genes:</p>
<pre class="r"><code>if(nrow(y) &gt; 0){
print(gseaplot(y, geneSetID =(as.data.frame(y))[1,1],title = as.character((as.data.frame(y))[1,2])));
print(cnetplot(y, categorySize = &quot;pvalue&quot;, foldChange = upentrez$log2FoldChange,colorEdge = T));
print(emapplot(y, showCategory = 15, color = &quot;p.adjust&quot;,layout = &quot;kk&quot;));
}</code></pre>
</div>
</div>
<div id="reactome-of-downregulated-genes" class="section level4">
<h4><span class="header-section-number">5.1.3.2</span> Reactome of downregulated genes</h4>
<pre class="r"><code>if(!is.null(x)){
if(min(x@result$p.adjust)&lt;0.05){
sprintf(&quot;Barplot for the most enriched reactome pathway for Kallisto/DESeq analysis for comparison %s vs %s&quot;, groups[1],groups[2]);
print(barplot(x, showCategory = 30));
sprintf(&quot;Dotplot for the most enriched reactome pathway for Kallisto/DESeq analysis for comparison %s vs %s&quot;, groups[1],groups[2]);
print(dotplot(x, showCategory=15));
sprintf(&quot;Enrichment map plot for the most enriched reactome pathway for Kallisto/DESeq downregulated genes analysis for comparison %s vs %s&quot;, groups[1],groups[2]);
print(emapplot(x, showCategory = 15, color = &quot;p.adjust&quot;,layout = &quot;kk&quot;));
print(cnetplot(x,showCategory = 5, categorySize = &quot;pvalue&quot;, foldChange = down_genes_list_k, node_label= F, colorEdge = T))}} else {
  print(&quot;NO enriched term found&quot;)
}</code></pre>
<p>GSEA of downregulated genes</p>
<pre class="r"><code>if(nrow(y) &gt; 0){
print(gseaplot(y, geneSetID =(as.data.frame(y))[1,1],title = as.character((as.data.frame(y))[1,2])));
print(cnetplot(y, categorySize = &quot;pvalue&quot;, foldChange = downentrez$log2FoldChange,colorEdge = T));
print(emapplot(y, showCategory = 15, color = &quot;p.adjust&quot;,layout = &quot;kk&quot;));
}
#downunique&lt;- down_genes_list[!duplicated(names(down_genes_list))]
#viewPathway(as.character((as.data.frame(y))[1,2]), readable = T, foldChange = downunique )</code></pre>
</div>
</div>
</div>
</div>
<div id="edger" class="section level1">
<h1><span class="header-section-number">6</span> EdgeR</h1>
<p>We can repeat the same analysis using edgeR as the inference engine (Robinson, McCarthy, and Smyth 2009). The following code incorporates the average transcript length matrix as an offset for an edgeR analysis.</p>
<p>But I will use the tximport imported counts again and calculate the nomalization.</p>
<pre class="r"><code>library(edgeR)
#I will use the data from Salmon imported with tximport
counts_kal&lt;-txikallisto$counts
normMat &lt;- txikallisto$length
normMat &lt;- normMat / exp(rowMeans(log(normMat)))
o &lt;- log(calcNormFactors(counts_kal/normMat)) + log(colSums(counts_kal/normMat))
EdgeR_k &lt;- DGEList(counts_kal)
EdgeR_k &lt;- scaleOffset(EdgeR_k, t(t(log(normMat)) + o))
keep &lt;- filterByExpr(EdgeR_k)
EdgeR_k &lt;- EdgeR_k[keep,]
#The basic edgeR model fitting and results extraction can be accomplished with the following lines:
EdgeR_k$samples$group&lt;-substr(colnames(txikallisto$counts),1,nchar(colnames(txikallisto$counts))-1)
EdgeR_k&lt;- calcNormFactors(EdgeR_k)
EdgeR_k&lt;-estimateDisp(EdgeR_k)</code></pre>
<p>Design matrix not provided. Switch to the classic mode. ## The results from the EdgeR</p>
<div id="counts-of-the-edgerdataset" class="section level3">
<h3><span class="header-section-number">6.0.1</span> Counts of the EdgeRdataset</h3>
<pre class="r"><code>head(as.data.frame(EdgeR_k$counts))</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":[""],"name":["_rn_"],"type":[""],"align":["left"]},{"label":["Goat_milk1"],"name":[1],"type":["dbl"],"align":["right"]},{"label":["Goat_milk2"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["Goat_milk3"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["Goat_milk4"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["No_milk1"],"name":[5],"type":["dbl"],"align":["right"]},{"label":["No_milk2"],"name":[6],"type":["dbl"],"align":["right"]},{"label":["No_milk3"],"name":[7],"type":["dbl"],"align":["right"]},{"label":["No_milk4"],"name":[8],"type":["dbl"],"align":["right"]}],"data":[{"1":"2237.0690","2":"2410.069","3":"1705.029","4":"2410.055","5":"834.000","6":"2013.056","7":"2936.056","8":"2085.000","_rn_":"ENSSSCG00000000002"},{"1":"2064.8956","2":"1840.028","3":"3706.352","4":"2531.433","5":"2896.035","6":"2725.737","7":"2525.139","8":"2648.602","_rn_":"ENSSSCG00000000003"},{"1":"220.0000","2":"314.000","3":"269.000","4":"327.000","5":"201.000","6":"264.000","7":"261.000","8":"338.000","_rn_":"ENSSSCG00000000005"},{"1":"515.0000","2":"160.000","3":"504.000","4":"320.000","5":"600.000","6":"627.000","7":"352.000","8":"327.000","_rn_":"ENSSSCG00000000006"},{"1":"535.0000","2":"632.000","3":"346.000","4":"583.000","5":"308.000","6":"566.000","7":"569.000","8":"516.000","_rn_":"ENSSSCG00000000007"},{"1":"899.7974","2":"1027.177","3":"2493.709","4":"2076.021","5":"1213.174","6":"1851.007","7":"1166.867","8":"1351.651","_rn_":"ENSSSCG00000000010"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<div id="plot-mds-of-the-data" class="section level3">
<h3><span class="header-section-number">6.0.2</span> Plot MDS of the data</h3>
<p>The difference between the PCA nad MDS is that PCR display the two dimentions with the largest variance in the data, MDS is based in the pairwise distances betwen the points of the data. They will be the same if the covariance in the data is equal to the euclidean distance between data points.</p>
<pre class="r"><code>plotMDS(EdgeR_k)</code></pre>
<p><img src="FiguresMDS_plot-1.png" /><!-- --></p>
</div>
<div id="de-analysis-using-exact-test" class="section level3">
<h3><span class="header-section-number">6.0.3</span> DE analysis using exact test</h3>
<p>I will use the exact test to do the pairwise comparison without any batch effect.</p>
<p>And the number of genes with pvalue &lt; 0.1</p>
<pre class="r"><code>EdgeR_k$samples$W1&lt;-set_k_n$W_1
design&lt;-model.matrix(as.formula(~ W1+ group),data =EdgeR_k$samples)
DE_fit&lt;-glmQLFit(EdgeR_k,design=design)
DE_single_factor_edgeR&lt;- glmQLFTest(DE_fit)
sig_DE_EdgeR_k&lt;-DE_single_factor_edgeR$table[DE_single_factor_edgeR$table$PValue &lt; 0.1,]
dim(sig_DE_EdgeR_k)</code></pre>
<p>[1] 858 4</p>
</div>
<div id="toptags" class="section level2">
<h2><span class="header-section-number">6.1</span> Toptags</h2>
<p>Number of Genes with log2FoldChange &lt; -1 or &gt; 1, and p.adjusted &lt; 0.1 and the table with the values.</p>
<pre class="r"><code>Topgenes_sfactor_edgeR_k&lt;-topTags(DE_single_factor_edgeR, n = nrow(sig_DE_EdgeR_k))
#The FDR is adj. Pvalue  
Topgenes_sfactor_edgeR_k_0.1_1&lt;- 
Topgenes_sfactor_edgeR_k[Topgenes_sfactor_edgeR_k$table$FDR &lt; 0.1   &amp; Topgenes_sfactor_edgeR_k$table$logFC &lt; -1 | Topgenes_sfactor_edgeR_k$table$logFC &gt; 1,]
dim(Topgenes_sfactor_edgeR_k_0.1_1)</code></pre>
<p>[1] 40 5</p>
<pre class="r"><code>head(Topgenes_sfactor_edgeR_k_0.1_1$table)</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":[""],"name":["_rn_"],"type":[""],"align":["left"]},{"label":["logFC"],"name":[1],"type":["dbl"],"align":["right"]},{"label":["logCPM"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["F"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["PValue"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["FDR"],"name":[5],"type":["dbl"],"align":["right"]}],"data":[{"1":"1.532227","2":"1.1913737","3":"33.76724","4":"0.0002101372","5":"0.9990476","_rn_":"ENSSSCG00000026943"},{"1":"1.289576","2":"3.7439875","3":"20.10505","4":"0.0013409510","5":"0.9990476","_rn_":"ENSSSCG00000027865"},{"1":"1.058842","2":"0.1815982","3":"14.55202","4":"0.0037539031","5":"0.9990476","_rn_":"ENSSSCG00000039537"},{"1":"1.327381","2":"-0.4331678","3":"12.82739","4":"0.0054481912","5":"0.9990476","_rn_":"ENSSSCG00000015766"},{"1":"1.241474","2":"1.3081369","3":"12.42029","4":"0.0059761723","5":"0.9990476","_rn_":"ENSSSCG00000008572"},{"1":"3.253249","2":"2.8995357","3":"12.13129","4":"0.0063890902","5":"0.9990476","_rn_":"ENSSSCG00000036203"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<pre class="r"><code>saveRDS(Topgenes_sfactor_edgeR_k_0.1_1$table, file = file.path(output_dir,sprintf(&quot;%s_vs_%s_EdgeR_sig_Kal_with_outliers.RDS&quot;,groups[1], groups[2])))</code></pre>
<div id="toptags-downregulated-genes" class="section level3">
<h3><span class="header-section-number">6.1.1</span> Toptags downregulated genes</h3>
<pre class="r"><code>#Get the down regulated genes in the second group
Topgenes_sfactor_edgeR_k_0.1_1_down&lt;- Topgenes_sfactor_edgeR_k_0.1_1[Topgenes_sfactor_edgeR_k_0.1_1$table$logFC &lt; 0, ]
Topgenes_sfactor_edgeR_k_0.1_1_down&lt;- Topgenes_sfactor_edgeR_k_0.1_1_down[order(Topgenes_sfactor_edgeR_k_0.1_1_down$table$logFC), ]
dim(Topgenes_sfactor_edgeR_k_0.1_1_down)</code></pre>
<p>[1] 0 5</p>
<pre class="r"><code>head(Topgenes_sfactor_edgeR_k_0.1_1_down$table)</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["logFC"],"name":[1],"type":["dbl"],"align":["right"]},{"label":["logCPM"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["F"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["PValue"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["FDR"],"name":[5],"type":["dbl"],"align":["right"]}],"data":[],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<div id="toptags-upregulated-genes" class="section level3">
<h3><span class="header-section-number">6.1.2</span> Toptags upregulated genes</h3>
<pre class="r"><code>#Getting the upregulated genes in the second group
Topgenes_sfactor_edgeR_k_0.1_1_up&lt;- Topgenes_sfactor_edgeR_k_0.1_1[Topgenes_sfactor_edgeR_k_0.1_1$table$logFC &gt; 0, ]
Topgenes_sfactor_edgeR_k_0.1_1_up&lt;-Topgenes_sfactor_edgeR_k_0.1_1_up[order(-Topgenes_sfactor_edgeR_k_0.1_1_up$table$logFC), ]
dim(Topgenes_sfactor_edgeR_k_0.1_1_up)</code></pre>
<p>[1] 40 5</p>
<pre class="r"><code>head(Topgenes_sfactor_edgeR_k_0.1_1_up$table)</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":[""],"name":["_rn_"],"type":[""],"align":["left"]},{"label":["logFC"],"name":[1],"type":["dbl"],"align":["right"]},{"label":["logCPM"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["F"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["PValue"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["FDR"],"name":[5],"type":["dbl"],"align":["right"]}],"data":[{"1":"3.321483","2":"7.067868","3":"3.381489","4":"0.09743211","5":"0.9990476","_rn_":"ENSSSCG00000012178"},{"1":"3.253249","2":"2.899536","3":"12.131293","4":"0.00638909","5":"0.9990476","_rn_":"ENSSSCG00000036203"},{"1":"3.197582","2":"7.475299","3":"3.551798","4":"0.09048593","5":"0.9990476","_rn_":"ENSSSCG00000026430"},{"1":"2.871486","2":"5.470407","3":"4.063256","4":"0.07305866","5":"0.9990476","_rn_":"ENSSSCG00000027046"},{"1":"2.379400","2":"4.958068","3":"6.943478","4":"0.02603284","5":"0.9990476","_rn_":"ENSSSCG00000037509"},{"1":"2.228667","2":"3.498815","3":"5.301778","4":"0.04542946","5":"0.9990476","_rn_":"ENSSSCG00000025253"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
</div>
<div id="creating-a-volcano-plot-and-a-heatmap" class="section level1">
<h1><span class="header-section-number">7</span> Creating a volcano plot and a heatmap</h1>
<pre class="r"><code>plot(DE_single_factor_edgeR$table$logFC,-log(DE_single_factor_edgeR$table$PValue),main =&quot;Volcano plot Kallisto data from EdgeR&quot;, xlab= &quot;Log2FoldChange&quot;, ylab = &quot;-log(Pvalue)&quot;, pch = 21, col=ifelse(DE_single_factor_edgeR$table$PValue&lt;0.001,&quot;red&quot;,&quot;black&quot;), xlim= c(-5,5), ylim =c(0,25), abline(v=c(-2,2)))</code></pre>
<p><img src="Figuresplot_DE_EdgeR_Kallisto-1.png" /><!-- --></p>
<pre class="r"><code>#Using normalized counts
counts_kal_n &lt;- cpm(EdgeR_k, normalized.lib.sizes = T)
Sig_counts&lt;-counts_kal_n[row.names(counts_kal_n) %in% row.names(Topgenes_sfactor_edgeR_k_0.1_1$table),]
var_genes &lt;- apply(Sig_counts, 1, var)
head(var_genes)</code></pre>
<p>ENSSSCG00000000789 ENSSSCG00000002981 ENSSSCG00000004159 1.110895e+01 2.816533e+04 2.625018e-01 ENSSSCG00000007083 ENSSSCG00000007605 ENSSSCG00000008572 8.197118e-01 5.931146e-01 1.977511e+00</p>
<pre class="r"><code>Sig_counts&lt;- cbind(Sig_counts, var_genes)
Sig_counts&lt;-Sig_counts[order(Sig_counts[,7],decreasing =T),]
len&lt;-ifelse(dim(Sig_counts)[1]&lt;50, dim(Sig_counts)[1],50)
heatmap.2(Sig_counts[1:len,1:(ncol(Sig_counts)-1)],  trace=&quot;none&quot;, 
              main=&quot;Most variable genes across samples Kallisto from EdgeR&quot;, scale=&quot;row&quot;,col = rev(morecols(50)),
              labCol = labels )</code></pre>
<p><img src="Figuresedger_heatmap-1.png" /><!-- --></p>
<div id="functional-analysis-of-the-edger-results" class="section level2">
<h2><span class="header-section-number">7.1</span> Functional analysis of the EdgeR results</h2>
<p>Using the DAVIDWEBServices to analize the DE genes and doing the plot2D of terms and evidence:</p>
<pre class="r"><code>if(is.connected(david)== FALSE) {
  connect(david)
}
DAVID_results_EdgeR_k&lt;- addList(david, row.names(Topgenes_sfactor_edgeR_k_0.1_1$table), idType = &quot;ENSEMBL_GENE_ID&quot; , listName = sprintf(&quot;%s_v_%s_Sig_genes_EdgeR&quot;,groups[1],groups[2]) ,listType = &quot;Gene&quot; )
# to see the p% of genes that were recognize on DAVID database
DAVID_results_EdgeR_k$inDavid</code></pre>
<p>[1] 0.55</p>
<pre class="r"><code>DAVID_results_chart&lt;- getFunctionalAnnotationChart(david)
DAVID_results_chart_EdgeR_k_BP&lt;-DAVID_results_chart[DAVID_results_chart$Category == &quot;GOTERM_BP_DIRECT&quot; &amp; DAVID_results_chart$PValue &lt; 0.05,]
if(dim(DAVID_results_chart_EdgeR_k_BP)[1]&gt;0){
plot2D(DAVIDFunctionalAnnotationChart(DAVID_results_chart_EdgeR_k_BP[1:5,]),
       color=c(&quot;FALSE&quot;=&quot;black&quot;, &quot;TRUE&quot;=&quot;green&quot;))
}else{
  print(&quot;No significant term found&quot;)
}</code></pre>
<p><img src="FiguresEdgeR_DAVID-1.png" /><!-- --></p>
<div id="comparing-the-results-from-deseq-and-edger" class="section level3">
<h3><span class="header-section-number">7.1.1</span> Comparing the results from DESeq and EdgeR</h3>
<pre class="r"><code>if(dim(DAVID_results_chart_DESeq_kal_BP) &amp; dim(DAVID_results_chart_EdgeR_k_BP)[1]&gt;0){
sprintf(&quot;DESeq2_%s_%s&quot;, groups[1],groups[2])
head(DAVID_results_chart_DESeq_kal_BP[order(DAVID_results_chart_DESeq_kal_BP$PValue),],10)
sprintf(&quot;EdgeR_%s_%s&quot;, groups[1],groups[2])
head(DAVID_results_chart_EdgeR_k_BP[order(DAVID_results_chart_EdgeR_k_BP$PValue),],10)
# We can see the top 5 GO pathways for biological process, and note how the evidence is true in our data for each of the genes in the pathway( if it is present on our sample or not)
#We can also use the chart to do a godag plot 
}else{
  print(&quot;Either DESeq or EdgeR functional chart is empty&quot;)
}</code></pre>
<pre><code>## Warning in if (dim(DAVID_results_chart_DESeq_kal_BP) &amp;
## dim(DAVID_results_chart_EdgeR_k_BP)[1] &gt; : the condition has length &gt; 1 and
## only the first element will be used</code></pre>
<p>[1] “Either DESeq or EdgeR functional chart is empty”</p>
</div>
</div>
<div id="significant-terms" class="section level2">
<h2><span class="header-section-number">7.2</span> Significant terms</h2>
<pre class="r"><code>DAVIDGODag_EdgeR_k&lt;-DAVIDGODag(DAVIDFunctionalAnnotationChart(DAVID_results_chart_EdgeR_k_BP), type =&quot;BP&quot;,pvalueCutoff=.001)
#we used a cutoff of 0.001 f or the pvalue. And now we can inspect the Dag object
sigCategories(DAVIDGODag_EdgeR_k, p=0.0001)</code></pre>
<p>NULL</p>
<pre class="r"><code>summary(DAVIDGODag_EdgeR_k)</code></pre>
<pre><code>## Warning: No results met the specified criteria. Returning 0-row data.frame</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["GOBPID"],"name":[1],"type":["chr"],"align":["left"]},{"label":["Pvalue"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["Count"],"name":[3],"type":["int"],"align":["right"]},{"label":["Size"],"name":[4],"type":["int"],"align":["right"]},{"label":["Term"],"name":[5],"type":["chr"],"align":["left"]}],"data":[],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<pre class="r"><code>show(DAVIDGODag_EdgeR_k)</code></pre>
<p>Gene to GO BP test for over-representation 0 GO BP ids tested (0 have p &lt; 0.001) Selected gene set size: 0</p>
<pre><code>## Warning in max(popTotals(r)): no non-missing arguments to max; returning -
## Inf</code></pre>
<pre><code>Gene universe size: -Inf 
Annotation package: DAVID.website </code></pre>
<pre class="r"><code># we can see by the FDR column which pathways are significant</code></pre>
<p>The graph plot of the EdgeR data</p>
<pre class="r"><code>plotGOTermGraph(g=goDag(DAVIDGODag_EdgeR_k), r= DAVIDGODag_EdgeR_k, max.nchar= 20, node.shape=&quot;ellipse&quot;,node.colors=c(sig=&quot;red&quot;, not=&quot;white&quot;))</code></pre>
<pre><code>## Warning in .local(x, y, ...): graph has zero nodes; cannot layout</code></pre>
</div>
</div>
<div id="comparison-between-the-deseq2-and-edger-results" class="section level1">
<h1><span class="header-section-number">8</span> Comparison between the DESeq2 and EdgeR results</h1>
<div id="venndiagram-of-total-genes" class="section level2">
<h2><span class="header-section-number">8.1</span> Venndiagram of total genes:</h2>
<pre class="r"><code>library(VennDiagram)
# Total DE genes 
I_total&lt;- intersect(rownames(resSig_Kal), rownames(Topgenes_sfactor_edgeR_k_0.1_1$table))
length(I_total)</code></pre>
<p>[1] 2</p>
<pre class="r"><code>grid.newpage()
draw.pairwise.venn(nrow(resSig_Kal), nrow(Topgenes_sfactor_edgeR_k_0.1_1$table), length(I_total), category = c(&quot;DESeq2&quot;, &quot;EdgeR&quot;), lty = rep(&quot;blank&quot;, 
    2), fill = c(&quot;light blue&quot;, &quot;pink&quot;), alpha = rep(0.5, 2), cat.pos = c(-15, 
    0), cat.dist = rep(0.025, 2), title= &quot;DE genes from DESeq and EdgeR&quot;)</code></pre>
<p><img src="Figuresintersect_DESeq_EdgeR-1.png" /><!-- -->(polygon[GRID.polygon.9891], polygon[GRID.polygon.9892], polygon[GRID.polygon.9893], polygon[GRID.polygon.9894], text[GRID.text.9895], text[GRID.text.9896], text[GRID.text.9897], lines[GRID.lines.9898], text[GRID.text.9899], text[GRID.text.9900])</p>
</div>
<div id="comparison-between-the-downregulated-genes" class="section level2">
<h2><span class="header-section-number">8.2</span> Comparison between the downregulated genes</h2>
<pre class="r"><code>I_down&lt;- intersect(rownames(downresSig_Kal), rownames(Topgenes_sfactor_edgeR_k_0.1_1_down$table))
length(I_down)</code></pre>
<p>[1] 0</p>
<pre class="r"><code>grid.newpage()
draw.pairwise.venn(nrow(downresSig_Kal), nrow(Topgenes_sfactor_edgeR_k_0.1_1_down$table), length(I_down), category = c(&quot;DESeq2&quot;, &quot;EdgeR&quot;), lty = rep(&quot;blank&quot;, 
    2), fill = c(&quot;light blue&quot;, &quot;pink&quot;), alpha = rep(0.5, 2), cat.pos = c(-15, 
    0), cat.dist = rep(0.025, 2), title= &quot;Down regulated genes according to DESeq and/or EdgeR&quot;)</code></pre>
<p><img src="Figuresvenn_down-1.png" /><!-- -->(polygon[GRID.polygon.9901], polygon[GRID.polygon.9902], polygon[GRID.polygon.9903], polygon[GRID.polygon.9904], text[GRID.text.9905], text[GRID.text.9906], text[GRID.text.9907], text[GRID.text.9908])</p>
</div>
<div id="comparison-between-the-upregulated-genes" class="section level2">
<h2><span class="header-section-number">8.3</span> Comparison between the upregulated genes</h2>
<pre class="r"><code>I_up&lt;- intersect(rownames(upresSig_Kal), rownames(Topgenes_sfactor_edgeR_k_0.1_1_up$table))
length(I_up)</code></pre>
<p>[1] 2</p>
<pre class="r"><code>grid.newpage()
draw.pairwise.venn(nrow(upresSig_Kal), nrow(Topgenes_sfactor_edgeR_k_0.1_1_up$table), length(I_up), category = c(&quot;DESeq2&quot;, &quot;EdgeR&quot;), lty = rep(&quot;blank&quot;, 
    2), fill = c(&quot;light blue&quot;, &quot;pink&quot;), alpha = rep(0.5, 2), cat.pos = c(-15, 
    0), cat.dist = rep(0.025, 2), title= &quot;up DE genes from DESeq and EdgeR&quot;)</code></pre>
<p><img src="Figuresvenn_up-1.png" /><!-- -->(polygon[GRID.polygon.9909], polygon[GRID.polygon.9910], polygon[GRID.polygon.9911], polygon[GRID.polygon.9912], text[GRID.text.9913], text[GRID.text.9914], lines[GRID.lines.9915], text[GRID.text.9916], lines[GRID.lines.9917], text[GRID.text.9918], text[GRID.text.9919])</p>
</div>
<div id="get-the-list-of-the-genes-that-are-in-both-deseq2-and-edger-de-genes." class="section level2">
<h2><span class="header-section-number">8.4</span> Get the list of the genes that are in both DESeq2 and edgeR DE genes.</h2>
<pre class="r"><code>head(I_total)</code></pre>
<p>[1] “ENSSSCG00000026943” “ENSSSCG00000027865”</p>
<pre class="r"><code>Kallisto_intersect&lt;-gene_names_removed_outliers_kal[rownames(gene_names_removed_outliers_kal) %in% I_total, ]
saveRDS(Kallisto_intersect,file = file.path(output_dir,sprintf(&quot;%s_vs_%s_Kallisto_intersect_with_outliers.RDS&quot;,groups[1], groups[2])))</code></pre>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
